vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflow.java

```diff
@@ -94,19 +81,32 @@ public void recover(String workflowName, String workflowVersion, String vidarrId
   }
 
   @Override
-  public void release(String workflowName, String workflowVersion, String vidarrId) {
+  public void release(String workflowName, String workflowVersion, String vidarrId, Optional<JsonNode> input) {
+
+    int workflowPriority = (int) acceptedPriorities.get(0);
+    if (!input.isEmpty()) {
+      workflowPriority = input.get().asInt();
+    }
+
+    SimpleEntry waiter = new SimpleEntry(vidarrId, workflowPriority);
+
+    if (workflows.get(workflowName) == null) {
+      set(workflowName, vidarrId, input);
+    }
+
     final var state = workflows.get(workflowName);
-    if (state != null) {
-      state.waiting.remove(vidarrId);
-      currentInPriorityWaitingCount.labels(workflowName).set(state.waiting.size());
+
+    if (!state.waiting.contains(waiter)){
+      state.waiting.add(waiter);
+
```

Comment:
It looks like `set` handles "maybe empty input", "maybe the workflow isn't already a member of `workflows`, and also updates the currentInPriorityWaitingCount` counter - what is the value to doing those steps separately here instead of letting `set` take care of things?

Reply:
you're right, I don't think there's any point duplicating those checks